### PR TITLE
Move to using assertThat() exclusively 

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description
The `assertThat()` style of assertions is more verbose but also more expressive.  The matchers provide a clear and consistent structure for describing actual and expected results. 

This change replaces all other `assert*` methods in the library with `assertThat(...)` methods and enforces the exclusive use of `assertThat()` verifications. 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time.

- [x] Push your changes to a branch other than `main`. Create your PR from that branch.
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [ ] Fill in the "Description" above.
- [ ] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by bitwiseman** [comment]
@gsmet @car-roll @timja 
This is a test only change meant to make tests more consistent and readable. Feedback welcome.

File: src/test/java/org/kohsuke/github/GHContentIntegrationTest.java [hunk 1]
```diff
@@ -58,31 +62,31 @@ public void testGetFileContent() throws Exception {
...
     }
 
     @Test
     public void testGetEmptyFileContent() throws Exception {
         GHContent content = repo.getFileContent("ghcontent-ro/an-empty-file");
 
-        assertTrue(content.isFile());
-        assertEquals("", content.getContent());
+        assertThat(content.isFile(), is(true));
+        assertThat(content.getContent(), equalTo(""));
```
Comment (timja) [inline src/test/java/org/kohsuke/github/GHContentIntegrationTest.java:74]: I think there's a matcher for this, roughly
```suggestion
        assertThat(content.getContent(), is(emptyString()));
```

File: src/test/java/org/kohsuke/github/GHIssueEventTest.java [hunk 2]
```diff
@@ -41,11 +43,11 @@ public void testEventsForSingleIssue() throws Exception {
     public void testRepositoryEvents() throws Exception {
         GHRepository repo = getRepository();
         List<GHIssueEvent> list = repo.listIssueEvents().toList();
-        assertTrue(list.size() > 0);
+        assertThat(list.size() > 0, is(true));
```
Comment (timja) [inline src/test/java/org/kohsuke/github/GHIssueEventTest.java:46]: this is a more normal way of doing it
```suggestion
        assertThat(list, is(not(empty()));
```

**Review by timja — APPROVED** [review]
I got about halfway through, couple of inline suggestions, no guarantees they compile

File: src/test/java/org/kohsuke/github/AbstractGitHubWireMockTest.java [hunk 3]
```diff
@@ -4,14 +4,10 @@
...
-import org.hamcrest.Description;
 import org.hamcrest.Matcher;
+import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
-import org.hamcrest.StringDescription;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Rule;
+import org.junit.*;
```
[the remark this answers is not in the document]
Comment (gsmet) [reply src/test/java/org/kohsuke/github/AbstractGitHubWireMockTest.java:10]: Yeah, I would avoid that too.

**Comment by gsmet** [comment]
It makes sense. Consistency is good even if it makes things a bit more verbose in the simple cases.

File: src/test/java/org/kohsuke/github/GHIssueEventTest.java (hunk 2 above)
Comment (bitwiseman) [reply src/test/java/org/kohsuke/github/GHIssueEventTest.java:46]: Yeah, there's a bunch of these kind of changes. I was thinking I could get by without them, but good call. I'll go through them once and pick off any easy ones like this.

**Review by timja — APPROVED** [review]

**Review by car-roll — APPROVED** [review]

File: src/test/java/org/kohsuke/github/AbstractGitHubWireMockTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (bitwiseman) [reply src/test/java/org/kohsuke/github/AbstractGitHubWireMockTest.java]: Good catch.

**Pull request merged**